packages/@aws-cdk/aws-cognito/lib/user-pool-email.ts

```diff
@@ -202,3 +203,59 @@ function encodeAndTest(input: string | undefined): string | undefined {
     return undefined;
   }
 }
+
+/**
+ * Formats `fromName` to comply RFC 5322
+ *
+ * @see https://www.rfc-editor.org/rfc/rfc5322#section-3.4
+ */
+function formatFromName(fromName: string): string {
```

Comment:
Thanks this is much easier to follow! The only question I have is how this
handles different types of quotes? Will it still work if you use `"` or ```

```ts
fromName: "'foo.bar'"
fromName: `'${someVar}'`
```

Reply:
Per RFC 5322, both the single quote `'` and the backquote ``` ` ``` are "atext". They are same as alphabet/digit letters. But surprising, the dot `.` is a "specials".

```
   atext           =   ALPHA / DIGIT /    ; Printable US-ASCII
                       "!" / "#" /        ;  characters not including
                       "$" / "%" /        ;  specials.  Used for atoms.
                       "&" / "'" /
                       "*" / "+" /
                       "-" / "/" /
                       "=" / "?" /
                       "^" / "_" /
                       "`" / "{" /
                       "|" / "}" /
                       "~"
```

Note: White space is not defined as "atext", but also allowed.

Examples:

| source | template | cause |
|-|-|-|
| `"'foo.bar'"` | quoted-string<br>`"From": "\"'foo.bar'\" <addr@example.com>"` | the dot `.` is "specials" |
| `"'foo bar'"` | bare<br>`"From": "'foo bar' <addr@example.com>"` | all chars are "atext" |
| `"Foo's Bar"` | bare<br>`"From": "Foo's Bar <addr@example.com>"` | all chars are "atext" |
| `'${price}'` | bare<br>`"From": "${price} <addr@example.com>"` | all chars are "atext" | 
| ``` "`foo bar`" ``` | bare<br>``` "From": "`foo bar` <addr@example.com>" ``` | all chars are "atext" |
| `'Café'` | mime encode<br>`"From": "=?UTF-8?B?Q2Fmw6k=?= <addr@example.com>"` | `é` is not US-ASCII |
| `'"Café"'` | entire `"Café"` including double quote `"` is mime encoded<br>`"From": "=?UTF-8?B?IkNhZsOpIg==?= <addr@example.com>"` | `é` is not US-ASCII |
| ``` `'${someVar}'` ``` | mime encode if `someVar` is not US-ASCII<br>quoted-string if `someVar` includes "specials"<br>otherwise, as-is | ``` `~` ``` is evaluated at runtime<br>single quote `'` is atext |
| ``` `"${someVar}"` ``` | mime encode if `someVar` is not US-ASCII<br>quoted-string again if `someVar` includes malformed escape<br>otherwise, as-is | already quoted-string form